### PR TITLE
added MCSUtil.scala for writing netcdf files from MCCNode object

### PR DESCRIPTION
**Pull request by kwhitehall**
Addressing issue #149

### CODE REVIEW

File: src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+ * limitations under the License.
+ */
+package org.dia.algorithms.mcc
+
+import java.util
+import scala.collection.mutable.ListBuffer
+import ucar.ma2.{ArrayInt, ArrayDouble, DataType}
+import ucar.nc2.{NetcdfFileWriter, Dimension, Attribute, Variable }
+import scala.reflect.ClassTag
+import scala.collection.JavaConversions._
```
Comment (rahulpalamuttam) [inline src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala]: Try using scala.collection.JavaConverters._ instead.
This forces users to use .asScala and .asJava when converting equivalent collections across the languages.
The implicit conversions that JavaConversions does can be a tad too much magical to trace.

File: src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+import scala.collection.mutable
+
+import org.dia.core.SciTensor
+import org.dia.tensors.AbstractTensor
+
```
Comment (rahulpalamuttam) [inline src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala:31]: The style checker checks the order of import statements.
Each import group is line separated.

Here is the template :

import java...

import scala....

import everything that's not spark or scispark

import org.apache.spark

import org.dia.... <-- us

Imports within a group follow alphabetical ordering.

File: src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        gridMap.foreach{case(k,v) => {
+            val indices = k.replace("(", "").replace(")","").replace(" ", "").split(",")
+            nodeGrid.setDouble(ima.set( (indices(0).toInt) - latMinOffset, (indices(1).toInt) - lonMinOffset), v.toInt)
+        }}
+    
+        nodeID = "/tmp/F"+thisNode.getFrameNum.toString+"CE"+thisNode.getCloudElemNum.toString.dropRight(2)+"_"+latMinOffset+"_"+latMaxOffset+"_"+lonMinOffset+"_"+lonMaxOffset+".nc"
+        netcdf_write(nodeID, nodeGrid, lats, lons)
+	
+	}
+}
```
Comment (rahulpalamuttam) [inline src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala]: The style checker expects a newline at the end of each file.

File: src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+		    if (!seenNodes.contains(second)){
+		        seenNodes += second
+		        val sNode = MCSNodeMap(second)
+		        extract_masked_data(sNode, lats, lons, tightestBox)  
+		    }  
+		})
+	}
+
+
+	/** Extract the node mask from the MCCNode metadata for the
```
Comment (rahulpalamuttam) [inline src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala]: The style checker uses javadoc style.
That is the the multiline comment asterisk is in line with the first asterisk at the top of the comment block.

For example :
https://gist.github.com/rahulpalamuttam/d0c89e8b700363e9deaf055e53942eb9

**Comment by rahulpalamuttam** [comment]
@kwhitehall 
This is awesome.
The style checker gave some suggestions for the file.

If you want to run it locally:
1. sbt scalastyle
2. sbt scalastyle:test --> for tests

I've added some inline comments.

File: src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala (hunk 1 above)
Comment (kwhitehall) [reply src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala]: Ack'ed.

**Comment by kwhitehall** [comment]
Thanks @rahulpalamuttam! Wasn't too sure how to do the linting on the files :) Will fix and update PR shortly.

File: src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala [hunk 5]
```diff
@@ -18,189 +18,169 @@
...
-import scala.collection.mutable.ListBuffer
-import ucar.ma2.{ArrayInt, ArrayDouble, DataType}
-import ucar.nc2.{NetcdfFileWriter, Dimension, Attribute, Variable }
-import scala.reflect.ClassTag
-import scala.collection.JavaConversions._
+
+import ucar.ma2.{ArrayDouble, ArrayInt, DataType}
+import ucar.nc2.{Attribute, Dimension, NetcdfFileWriter, Variable}
+
+import scala.collection.JavaConverters._
```
Comment (rahulpalamuttam) [inline src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala:22]: Nice use of JavaConverters.

The ordering can be fixed by swapping the import.scala... statements and import ucar.ma2... statements.

The group ordering is as follows :

import java...

import scala....

import (everything else that's not org.apache.spark..., org.dia...., java..., or scala...)

org.apache.spark....

org.dia....

**Comment by kwhitehall** [comment]
Ack'ed @rahulpalamuttam. This is done. Thank you for the direction on this!

**Comment by rahulpalamuttam** [comment]
Awesome @kwhitehall.
This looks good to me.
The coverage decreased since the PR didn't have corresponding tests.
We can add appropriate tests later, unless you think it is better to tackle right now.

**Comment by kwhitehall** [comment]
Ack'ed @rahulpalamuttam. I say add test later as we need to add for the objects (node & edge) as well, and this will build off of those.
